File: WordPig/app/src/main/java/com/pigdogbay/wordpig/model/Tray.java
package com.pigdogbay.wordpig.model;

import android.util.Log;

import com.pigdogbay.wordpig.Defines;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;
import java.util.List;

/**
 * Created by Mark on 07/04/2015.
 */
public class Tray
{
    private StringBuilder buffer;
    private List<Tile> trayTiles;

    public Tray()
    {
        trayTiles = new ArrayList<Tile>();
        buffer = new StringBuilder();
    }

    public void setTilesIfInTray(List<Tile> tiles)
    {
        trayTiles.clear();
        for (Tile t : tiles)
        {
            if (t.y> Defines.TRAY_MIN_Y && t.y<Defines.TRAY_MAX_Y){
                trayTiles.add(t);
            }
        }
    }

    public void sortTilesByPosition()
    {
        Collections.sort(trayTiles, new Comparator<Tile>() {
            @Override
            public int compare(Tile lhs, Tile rhs) {
                return lhs.x - rhs.x;
            }
        });
    }

    public int getScore()
    {
        int score = 0;
        for (Tile t : trayTiles)
        {
            int index = t.letter-'a';
            score = score + Defines.SCORE_LETTER[index];
        }
        return score;
    }

    @Override
    public String toString() {
        buffer.setLength(0);
        for (Tile t : trayTiles)
        {
            buffer.append((char) t.letter);
        }
        return buffer.toString();
    }
}

File: WordPig/app/src/main/java/com/pigdogbay/wordpig/model/Tray.kt
package com.pigdogbay.wordpig.model

import com.pigdogbay.wordpig.Defines

class Tray {
    private val buffer = StringBuilder()
    private val trayTiles = ArrayList<Tile>()

    fun setTilesIfInTray(tiles: List<Tile>) {
        trayTiles.clear()
        for (t in tiles) {
            if (t.y > Defines.TRAY_MIN_Y && t.y < Defines.TRAY_MAX_Y) {
                trayTiles.add(t)
            }
        }
    }

    fun sortTilesByPosition() {
        trayTiles.sortWith { lhs, rhs -> lhs.x - rhs.x }
    }

    val score: Int
        get() {
            var score = 0
            for (t in trayTiles) {
                val index = t.letter - 'a'.code
                score += Defines.SCORE_LETTER[index]
            }
            return score
        }

    override fun toString(): String {
        buffer.setLength(0)
        for (t in trayTiles) {
            buffer.append(t.letter.toChar())
        }
        return buffer.toString()
    }
}
